src/Header/RequestHandler.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef GROUP4_HEADER_REQUESTHANDLER_H_
+#define GROUP4_HEADER_REQUESTHANDLER_H_
+
+#include "OperateData.h"
+#include "Authenticator.h"
+#include "APIHandler.h"
+#include <string>
+#include <unordered_map>
+#include <memory>
+
+//Will decode the request string and check if it's a valid request.
+//calling RequestParser to get OperateData then call correspond api function
+class RequestHandler
+{
+public:
+    //default constructor will never be used
+    RequestHandler();
+    //instantiate instance with request string, will parse to OperateData and check with Authenticator and call correspond api.
+    RequestHandler(const std::string &requestData);
+    //data is content of request body
+    std::unique_ptr<OperateData> data;
```

Comment:
If I understand correctly, OperateData represents the entire HTTP request. If that's the case, you probably don't need to separate data from header.

Reply:
Done